sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -53,6 +70,27 @@ def test_feature_hasher_pairs():
     assert_equal([1, 3, 4], x2_nz)
 
 
+def test_feature_hasher_pairs_with_string_values():
+    raw_X = (iter(d.items()) for d in [{"foo": 1, "bar": "a"},
+                                       {"baz": u"abc", "quux": 4, "foo": -1}])
+    h = FeatureHasher(n_features=16, input_type="pair")
+    x1, x2 = h.transform(raw_X).toarray()
+    x1_nz = sorted(np.abs(x1[x1 != 0]))
+    x2_nz = sorted(np.abs(x2[x2 != 0]))
+    assert_equal([1, 1], x1_nz)
+    assert_equal([1, 1, 4], x2_nz)
+
+    raw_X = (iter(d.items()) for d in [{"foo": "bar", "bar": "a"},
```

Comment:
It's not clear what this is testing that was not in the previous part of the test.